benchmarking/benchmark.py

```diff
@@ -0,0 +1,346 @@
+            start = time()
+            X_transformed = emf_transformer.transform(subset)
+            end = time()
+            times[i] = end - start
+        result.append(sum(times) / N_REPEATS)
+    return np.array(result)
+
+
+def get_generator_times_rdkit(X, n_molecules, generator, count, sparse):
+    if count:
+        fp_function = lambda x: generator.GetCountFingerprint(
+            MolFromSmiles(x)
+        ).ToList()
+    else:
+        fp_function = lambda x: generator.GetFingerprint(MolFromSmiles(x))
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array([fp_function(x) for x in subset])
+            else:
+                X_transformed = np.array([fp_function(x) for x in subset])
+            end = time()
+            times[i] = end - start
+        result.append(sum(times) / N_REPEATS)
+    return np.array(result)
+
+
+def get_times_rdkit(X, n_molecules, func, sparse=False, **kwargs):
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            else:
+                X_transformed = np.array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            end = time()
+            times[i] = end - start
+        result.append(sum(times) / N_REPEATS)
+    return np.array(result)
+
+
+def plot_results(
+    n_molecules, y_emf, y_rdkit, title="", sparse=None, count=None
+):
+    if sparse is not None:
+        if sparse:
+            title += " sparse"
+
+    if count is not None:
+        if count:
+            title += " count"
+        else:
+            title += " bit"
+
+    X = n_molecules * np.linspace(0, 1, N_SPLITS + 1)[1:]
+
+    plt.rcParams["font.size"] = 20
+    fig = plt.figure(figsize=(15, 10))
+    ax1 = fig.add_subplot()
+    ax1.set_title(title)
+
+    ax1.plot(X, y_emf[0], label="emf time - 1 job")
+    ax1.plot(X, y_emf[1], label="emf time - 2 job")
+    ax1.plot(X, y_emf[2], label="emf time - 4 job")
+    ax1.plot(X, y_emf[3], label="emf time - all jobs")
+    ax1.plot(X, y_rdkit, label="rdkit time")
+
+    ax1.set_ylabel("Time of computiation")
+    ax1.set_xlabel("Number of finberprints")
+
+    ax1.set_xlim(n_molecules * 0.1, n_molecules * 1.1)
+    ax1.set_ylim(bottom=0)
+
+    plt.legend(loc="upper left")
+    plt.savefig(title.replace(" ", "_") + ".png")
+    # plt.show()
+
+
+from rdkit.Chem import MolFromSmiles
```

Comment:
Can't this be moved to the top of the file?

Reply:
Done